test/kinesis/custom-endpoint.test.js

```diff
@@ -0,0 +1,31 @@
+const Environment = require('../../src/Environment');
+const test = require('ava');
+const sinon = require('sinon');
+const AWS = require('aws-sdk');
+const AWSMock = require('aws-sdk-mock');
+AWSMock.setSDKInstance(AWS);
+
+// Mock listTables for `beforeAll` and `afterEach` hooks
+const listStreams = sinon.stub()
+  .resolves({ StreamNames: [] }, []);
+
+AWSMock.mock('Kinesis', 'listStreams', listStreams);
+
+const { useKinesisDocker } = require('../../src/kinesis');
+const environment = new Environment();
+
+test.before(() => {
+  environment.set('AWS_ACCESS_KEY_ID', 'test-access-key');
+  environment.set('AWS_SECRET_ACCESS_KEY', 'test-secret-key');
+  environment.set('KINESIS_ENDPOINT', 'kinesis://localhost');
+});
+
+test.after(() => environment.restore());
+
+useKinesisDocker(test);
+
+test.serial('When KINESIS_ENDPOINT is set configuration environment variables are not set', (test) => {
```

Comment:
I would also expect that when `KINESIS_ENDPOINT` is set a docker container is not started and the resulting client uses the referenced Kinesis instance. Could we add a test for this?

Reply:
I added a validation for this.